internal/operands/vm-console-proxy/reconcile_test.go

```diff
@@ -35,49 +38,187 @@ const (
+
+		updatedService := &core.Service{}
+		Expect(request.Client.Get(request.Context, key, updatedService)).ToNot(HaveOccurred())
+		Expect(updatedService.Spec.ClusterIP).To(Equal(testClusterIp))
+	})
+
+	It("should report status", func() {
+		reconcileResults, err := operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Set status for deployment
+		key := client.ObjectKeyFromObject(&bundle.Deployment)
+		updateDeployment(key, &request, func(deployment *apps.Deployment) {
+			deployment.Status.Replicas = replicas
+			deployment.Status.ReadyReplicas = 0
+			deployment.Status.AvailableReplicas = 0
+			deployment.Status.UpdatedReplicas = 0
+			deployment.Status.UnavailableReplicas = replicas
+		})
+
+		reconcileResults, err = operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Only deployment should be progressing
+		for _, reconcileResult := range reconcileResults {
+			if _, ok := reconcileResult.Resource.(*apps.Deployment); ok {
+				Expect(reconcileResult.Status.NotAvailable).ToNot(BeNil())
+				Expect(reconcileResult.Status.Progressing).ToNot(BeNil())
+				Expect(reconcileResult.Status.Degraded).ToNot(BeNil())
+			} else {
+				Expect(reconcileResult.Status.NotAvailable).To(BeNil())
+				Expect(reconcileResult.Status.Progressing).To(BeNil())
+				Expect(reconcileResult.Status.Degraded).To(BeNil())
+			}
+		}
+
+		updateDeployment(key, &request, func(deployment *apps.Deployment) {
+			deployment.Status.Replicas = replicas
+			deployment.Status.ReadyReplicas = replicas
+			deployment.Status.AvailableReplicas = replicas
+			deployment.Status.UpdatedReplicas = replicas
+			deployment.Status.UnavailableReplicas = 0
+		})
+
+		reconcileResults, err = operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		// All resources should be available
+		for _, reconcileResult := range reconcileResults {
+			Expect(reconcileResult.Status.NotAvailable).To(BeNil())
+			Expect(reconcileResult.Status.Progressing).To(BeNil())
+			Expect(reconcileResult.Status.Degraded).To(BeNil())
+		}
+	})
+
+	It("should deploy resources in namespace provided by annotation", func() {
+		_, err := operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		namespace := request.Instance.GetAnnotations()[VmConsoleProxyNamespaceAnnotation]
+		ExpectNamespacedResourceExists(&bundle.ServiceAccount, request, namespace)
+		ExpectNamespacedResourceExists(&bundle.ConfigMap, request, namespace)
+		ExpectNamespacedResourceExists(&bundle.Service, request, namespace)
+		ExpectNamespacedResourceExists(&bundle.Deployment, request, namespace)
+		ExpectNamespacedResourceExists(newRoute(getVmConsoleProxyNamespace(&request), serviceName), request, namespace)
+	})
+
+	It("should delete resources when enabled annotation is removed", func() {
+		_, err := operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		request.Instance.Annotations = map[string]string{
+			VmConsoleProxyNamespaceAnnotation: namespace,
+		}
```

Comment:
Instead of setting the annotation map, can you just delete the `EnableAnnotation`?
```go
delete(request.Instance.Annotations, EnableAnnotation)
```

Then it will be more obvious, that this test removes an annotation.

Reply:
Done, thanks!